e2e/volumegroupsnapshot_base.go

```diff
@@ -0,0 +1,427 @@
+}
+
+func (v volumeGroupSnapshotterBase) DeleteVolumeGroupSnapshotClass(groupSnapshotClassName string) error {
+	ctx := context.TODO()
+	err := v.groupclient.VolumeGroupSnapshotClasses().Delete(
+		ctx, groupSnapshotClassName, metav1.DeleteOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to delete VolumeGroupSnapshotClass: %w", err)
+	}
+	start := time.Now()
+	framework.Logf("Waiting for VolumeGroupSnapshotClass %v to be deleted", groupSnapshotClassName)
+	timeout := time.Duration(v.timeout) * time.Minute
+
+	return wait.PollUntilContextTimeout(
+		ctx,
+		poll,
+		timeout,
+		true,
+		func(ctx context.Context) (bool, error) {
+			_, err := v.groupclient.VolumeGroupSnapshotClasses().Get(ctx, groupSnapshotClassName, metav1.GetOptions{})
+			if err != nil {
+				if isRetryableAPIError(err) {
+					return false, nil
+				}
+				if apierrs.IsNotFound(err) {
+					return true, nil
+				}
+				framework.Logf("%s VolumeGroupSnapshotClass to be deleted (%d seconds elapsed)",
+					groupSnapshotClassName,
+					int(time.Since(start).Seconds()))
+
+				return false, fmt.Errorf("failed to get VolumeGroupSnapshotClass: %w", err)
+			}
+
+			return false, nil
+		})
+}
+
+func (v *volumeGroupSnapshotterBase) testVolumeGroupSnapshot(vol VolumeGroupSnapshotter) error {
+	pvcLabels := map[string]string{"pvc": "vgsc"}
+	pvcs, err := v.CreatePVCs(v.namespace, pvcLabels)
+	if err != nil {
+		return fmt.Errorf("failed to create PVCs: %w", err)
+	}
+
+	vgsc, err := vol.GetVolumeGroupSnapshotClass()
+	if err != nil {
+		return fmt.Errorf("failed to get volume group snapshot class: %w", err)
+	}
+	// Create a volume group snapshot class
+	vgscName := v.framework.Namespace.Name + "-vgsc"
+	vgsc.Name = vgscName
+	err = v.CreateVolumeGroupSnapshotClass(vgsc)
+	if err != nil {
+		return fmt.Errorf("failed to create volume group snapshot: %w", err)
+	}
+	vgsName := v.framework.Namespace.Name + "-vgs"
+	// Create a volume group snapshot
+	volumeGroupSnapshot, err := v.CreateVolumeGroupSnapshot(vgsName, vgscName, pvcLabels)
+	if err != nil {
+		return fmt.Errorf("failed to create volume group snapshot: %w", err)
+	}
+
+	clonePVCs, err := v.CreatePVCClones(volumeGroupSnapshot)
+	if err != nil {
+		return fmt.Errorf("failed to create clones: %w", err)
+	}
+	// validate the resources in the backend
+	err = vol.ValidateResourcesForCreate(volumeGroupSnapshot)
+	if err != nil {
+		return fmt.Errorf("failed to validate resources for create: %w", err)
+	}
+
+	// Delete the clones
+	err = v.DeletePVCs(clonePVCs)
+	if err != nil {
+		return fmt.Errorf("failed to delete clones: %w", err)
+	}
+	// Delete the PVCs
+	err = v.DeletePVCs(pvcs)
+	if err != nil {
+		return fmt.Errorf("failed to delete PVCs: %w", err)
+	}
+	// Delete the volume group snapshot
+	err = v.DeleteVolumeGroupSnapshot(volumeGroupSnapshot.Name)
+	if err != nil {
+		return fmt.Errorf("failed to delete volume group snapshot: %w", err)
+	}
+	// validate the resources in the backend after deleting the resources
```

Comment:
delete VolumeGroupSnapshotClass as well

Reply:
good point, it was missing, added it now